### PR TITLE
Feature/253 quality checks

### PR DESCRIPTION
**Pull request by lukaszdrynski**
https://triadmoj.atlassian.net/browse/DIV-253

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ node_modules/*
 *.swo
 .idea/
 *.iml
+npm-debug.log
```
Comment (sophiedrake) [inline .gitignore]: maybe just `*.log`?

File: app/routes.js [hunk 2]
```diff
@@ -18,406 +18,407 @@ router.get('/check-session', function (req, res) {
...
-     sandboxUrl: config.sanboxUrl,
-     herokuUrl: config.herokuUrl,
-     herokuV2Url: config.herokuV2Url,
-     herokuV1Url: config.herokuV1Url,
-   }
+  'index', {
+    sandboxUrl: config.sanboxUrl,
+    herokuUrl: config.herokuUrl,
+    herokuV2Url: config.herokuV2Url,
+    herokuV1Url: config.herokuV1Url,
```
Comment (sophiedrake) [inline app/routes.js]: Is this extra comma?

File: app/routes.js [hunk 3]
```diff
@@ -18,406 +18,407 @@ router.get('/check-session', function (req, res) {
...
 
-    res.redirect('../about-you/details');
-  });
+  if (req.session.respondent.homeAddress === 'No') {
+    res.redirect('address/postcode');
+  } else {
+    res.redirect('../reason-for-divorce');
+  }
+});
+router.post('/about-your-marriage/in-the-uk', function (req, res) {
```
Comment (sophiedrake) [inline app/routes.js]: Add new line above

File: app/routes.js [hunk 4]
```diff
@@ -18,406 +18,407 @@ router.get('/check-session', function (req, res) {
...
+    res.redirect('account');
+  }
+});
 
-    if (wishToName === 'Yes') {
-      res.redirect('name-person');
-    } else {
-      res.redirect('../../about-your-marriage/certificate');
-    }
-  });
```
Comment (sophiedrake) [inline app/routes.js]: Remove extra line

File: app/routes.js [hunk 5]
```diff
@@ -18,406 +18,407 @@ router.get('/check-session', function (req, res) {
...
+router.post('/statement-of-case/adultery/address/postcode', function (req, res) {
+  req.session.adultery.corespondent.postcode = req.body.postcode;
+  res.redirect('select');
+});
+
+router.post('/statement-of-case/adultery/address/select', function (req, res) {
+  req.session.adultery.corespondent.address = req.body.address;
+  res.redirect('../../../about-your-marriage/certificate');
+});
+router.post('/statement-of-case/adultery/address/enter-manually', function (req, res) {
```
Comment (sophiedrake) [inline app/routes.js]: Add new line above

File: app/routes.js [hunk 6]
```diff
@@ -429,397 +430,396 @@ router.get('/about-you/address/postcode', function (req, res) {
...
-  router.post('/statement-of-case/adultery/address/abroad', function (req, res) {
-    res.redirect('../../../about-your-marriage/certificate');
-  });
+router.post('/respondent/address/abroad', function (req, res) {
+  res.redirect('../../reason-for-divorce');
+});
 
+router.post('/statement-of-case/adultery/address/abroad', function (req, res) {
+  res.redirect('../../../about-your-marriage/certificate');
+});
```
Comment (sophiedrake) [inline app/routes.js]: Remove extra line

File: server.js [hunk 7]
```diff
@@ -197,9 +196,9 @@ app.get(/^\/([^.]+)$/, function (req, res) {
 
 });
 
-console.log("\nGOV.UK Prototype kit v" + releaseVersion);
+logger.info('\nGOV.UK Prototype kit v' + releaseVersion);
 // Display warning not to use kit for production services.
-console.log("\nNOTICE: the kit is for building prototypes, do not use it for production services.");
+logger.info('\nNOTICE: the kit is for building prototypes, do not use it for production services.');
```
Comment (sophiedrake) [inline server.js]: Where does this logger output to?

File: start.js [hunk 8]
```diff
@@ -1,15 +1,19 @@
...
+  logger.error('ERROR: Node module folder missing. Try running `npm install`');
   process.exit(0);
 }
 
 // remove .port.tmp if it exists  
 try {
   fs.unlinkSync(__dirname + '/.port.tmp');
-} catch(e){}
+} catch(e){
+  //logger.error(e);
```
Comment (sophiedrake) [inline start.js]: not logging the error here?

**Comment by sophiedrake** [comment]
Is it worth updating either the readme or wiki (not sure which we are using) with what checks we are running and how/when the checks are run?

File: server.js (hunk 7 above)
Comment (lukaszdrynski) [inline server.js]: stdout, same as console.log

**Comment by sophiedrake** [comment]
+1

**Pull request merged**